### PR TITLE
Respect SPHINXOPTS and add --install-deps flags to `spin docs`

### PR DESCRIPTION
**Pull request by stefanv**
@lagru

### CODE REVIEW

File: .spin/cmds.py [hunk 1]
```diff
@@ -9,9 +9,21 @@
...
 @click.option(
-    "--clean", is_flag=True, help="Clean previously built docs before building"
+    "--clean", is_flag=True,
+    default=False,
+    help="Clean previously built docs before building"
 )
-def docs(clean=False):
+@click.option(
+    "--raise-on-warning/--no-raise-on-warning",
+    default=True,
```
Comment (lagru) [inline .spin/cmds.py:18]:
```suggestion
    default=True,
    envvar="SPIN_RAISE_ON_WARNING",
```

File: .spin/cmds.py [hunk 2]
```diff
@@ -9,9 +9,21 @@
...
+    help="Convert warnings to errors"
+)
+@click.option(
+    "--install-deps/--no-install-deps",
+    default=True,
```
Comment (lagru) [inline .spin/cmds.py:23]:
```suggestion
    default=True,
    envvar="SPIN_AUTOINSTALL_DOC_DEPS",
```
Might be useful for those who want to change the default behavior locally...

File: .spin/cmds.py [hunk 3]
```diff
@@ -9,9 +9,21 @@
...
 @click.command()
 @click.option(
-    "--clean", is_flag=True, help="Clean previously built docs before building"
+    "--clean", is_flag=True,
+    default=False,
+    help="Clean previously built docs before building"
 )
-def docs(clean=False):
+@click.option(
+    "--raise-on-warning/--no-raise-on-warning",
```
Comment (lagru) [inline .spin/cmds.py:17]: Might be worth turning this option into `--sphinx-opts` that accepts a string with the default being `'-W'`?

**Review by lagru — APPROVED** [review]
Thanks. Fee free to merge with or without suggestions. :+1:

**Comment by stefanv** [comment]
@lagru Preferably, there should be one obvious way to set these options. Since Sphinx relies on `SPHINXOPTS`, I've removed that option and now no longer override `SPHINXOPTS` blindly. I've also added an example of how to use it.

Since the `--no-install-deps` is a tool option, I prefer to leave that explicit..

**Comment by lagru** [comment]
> removed that option and now no longer override SPHINXOPTS blindly.

Perfect, that seems indeed like the better route. 

> Since the --no-install-deps is a tool option, I prefer to leave that explicit.

Honestly and personally, I've found this behavior mostly annoying in my workflow. That's why I suggested adding `env="SPIN_AUTOINSTALL_DOC_DEPS"` flag as a compromise. But maybe it's a good thing that it forces me to update requirement files before building docs. 

By the way, should we aim to use spin in our CI as well? We are spending our maintenance time on it anyway and this way we can hopefully avoid errors due to differing tools locally and in the CI...

**Comment by stefanv** [comment]
> Honestly and personally, I've found this behavior mostly annoying in my workflow. That's why I suggested adding `env="SPIN_AUTOINSTALL_DOC_DEPS"` flag as a compromise. But maybe it's a good thing that it forces me to update requirement files before building docs.

Sorry, I got stuck on SPHINXOPTS; I think it'd be okay to add an environment variable for this option, if you find that helpful. I can't think of another way to make this much easier. We should just choose a convenient naming scheme, and then use it throughout. The docs on environment variable handling are somewhat lacking; we may want to customize it a bit, because currently, e.g., an empty environment variable is simply ignored.

> By the way, should we aim to use spin in our CI as well? We are spending our maintenance time on it anyway and this way we can hopefully avoid errors due to differing tools locally and in the CI...

The most relevant place to do this would be the building from source target. If you submitted a PR to make that use spin, I'd be happy to review.

**Pull request merged**